recipe_app/views.py

```diff
@@ -0,0 +1,179 @@
+import json
+import emoji
+
+from typing import Optional, Dict
+
+from django.http import HttpRequest
+from django.http import JsonResponse
+from django.http import Http404
+from django.db import transaction
+from django.db.models import QuerySet
+
+from rest_framework.views import APIView
+from rest_framework.parsers import JSONParser
+
+# Not stable for production, but makes testing easier
+from django.utils.decorators import method_decorator
+from django.views.decorators.csrf import csrf_exempt
+
+from .models import Recipe, Ingredient
+from .serializers import RecipeSerializer
+
+
+# Make the view exempt from CSRF checks
+# ! Not secure, just used for testing purposes
+@method_decorator(csrf_exempt, name="dispatch")
+class RecipeView(APIView):
```

Comment:
You could also consider a [ModelViewSet](https://www.django-rest-framework.org/api-guide/viewsets/#modelviewset) here which takes care of the boilerplate stuff for you

Reply:
That's a good find, I'll have a look at refactoring to utilise the framework a bit more.